src/io/dataset.cpp

```diff
@@ -1399,6 +1391,8 @@ void Dataset::AddFeaturesFrom(Dataset* other) {
   PushVector(&group_feature_cnt_, other->group_feature_cnt_);
   PushVector(&forced_bin_bounds_, other->forced_bin_bounds_);
   feature_groups_.reserve(other->feature_groups_.size());
+  // FIXME: fix the multiple multi-val feature groups, they need to be merged
+  // into one multi-val group
```

Comment:
@guolinke Should a new separate issue be created for this to not lose it?

Reply:
Already created.

Reply:
Ah, you mean #2754. Due to we have a lot of PRs modifying the same files, I overlooked it. Sorry about that!